ni/github.py

```diff
@@ -174,15 +181,15 @@ def __init__(self, server: ni_abc.ServerHost, client: aiohttp.ClientSession,
         cla_labels.sort()
         return cla_labels[0] if len(cla_labels) > 0 else None
 
-    async def set_label(self, status: ni_abc.Status) -> str:
+    async def set_label(self, problems: Mapping[ni_abc.Status, AbstractSet[str]]) -> str:
         """Set the label on the pull request based on the status of the CLA."""
         labels_url = await self.labels_url()
-        if status == ni_abc.Status.signed:
-            await self._gh.post(labels_url, data=[CLA_OK])
-            return CLA_OK
-        else:
+        if problems:
```

Comment:
This is a good example of why `defaultdict` would be troublesome because if I had simply checked for `Status.not_signed` then this would have been considered true even if the set for `Status.not_signed` was empty.